src/java.base/share/classes/java/lang/invoke/MethodHandles.java

```diff
@@ -3525,6 +3525,34 @@ public MethodHandle unreflectConstructor(Constructor<?> c) throws IllegalAccessE
             return lookup.getDirectConstructorNoSecurityManager(ctor.getDeclaringClass(), ctor);
         }
 
+        /*
+         * Produces a method handle that is capable of creating instances of the given class
+         * and instantiated by the given constructor.  No security manager check.
+         *
+         * This method is only used to implement the unsupported
+         * sun.reflect.ReflectionFactory::newConstructorForSerialization API.
+         */
```

Comment:
The following is probably more correct:
```suggestion
        /**
         * Produces a method handle that is capable of creating instances of the given class
         * and instantiated by the given constructor.  No security manager check.
         *
         * This method is used to implement serialization and the unsupported
         * sun.reflect.ReflectionFactory::newConstructorForSerialization API.
         */
```

Reply:
What are you referring to "to implement serialization"?   It's only used by ReflectionFactory.

Reply:
It’s called from the implementation of `JavaLangInvokeAccess::serializableConstructor`, which is called by `jdk.internal.reflect.ReflectionFactory::newConstructorForSerialization`, which is called by `java.io.ObjectStreamClass::getSerializableConstructor`:
https://github.com/openjdk/jdk/blob/86a18f5e2e0825dddb77656b2f43f64684f1464c/src/java.base/share/classes/java/io/ObjectStreamClass.java#L1444-L1446

Reply:
I see the confusion.   `newConstructionForSerialization` is used by serialization, exposed via `sun.reflect.ReflectionFactory`.   I may clarify it.